### PR TITLE
Refactors + prototype client: it is almost 6am

### PR DESCRIPTION
**Pull request by bobheadxi**
:tickets: **Ticket(s)**: n/a

---

## :construction_worker: Changes

- Big, big, massive refactors to core, mostly around code formatting, a lot around test structures
- Add Verification stuff to the database
- Improve email verification process
- Add `pinpoint` client as dependency of the web app
- Prototype API usage in frontend

## :flashlight: Testing Instructions

```
make testenv
make core
make gateway
make web
```

then go to localhost:3000/login

type anything

youll get status 200, tada (no user validation on the backend yet oops)

### CODE REVIEW

File: core/service/service.go [hunk 1]
```diff
@@ -159,10 +164,9 @@ func (s *Service) Handshake(ctx context.Context, req *request.Empty) (*response.
 	return &response.Empty{}, nil
 }
 
-// CreateAccount sends an email verification email. TODO: Actually create account
+// CreateAccount resgiesters a user and sends an email verification email
```
Comment (leesw98) [inline core/service/service.go]: `resgiesters ` 👍

File: core/crypto/creds.go [hunk 2]
```diff
@@ -51,16 +47,6 @@ func ValidateCredentialValues(usernames []string, password string) error {
 	return nil
 }
 
-// isLegalUserName returns true if the chosen username only contains characters [A-Z], [a-z], or '_' or '-'
-func isLegalUserName(username string) bool {
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply core/crypto/creds.go]: usernames are arbitrary, i think its safe to assume AWS will be immune against garbage input

[the remark this answers is not in the document]
Comment (bobheadxi) [reply core/crypto/creds.go]: so im fine with allowing emoji and symbols in names

[the remark this answers is not in the document]
Comment (bobheadxi) [reply core/crypto/creds.go]: its just a user-facing identifier

[the remark this answers is not in the document]
Comment (leesw98) [reply core/crypto/creds.go]: Okkie

**Pull request merged**